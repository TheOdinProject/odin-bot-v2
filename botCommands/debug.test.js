const command = require('./debug');

describe('/debug', () => {
  it.each([
    ['/debug'],
    [' /debug'],
    ['/debug @odin-bot'],
    [' /debug @odin-bot'],
    ['/debug js'],
    ['/debug rb'],
    [' /debug js'],
    [' /debug rb'],
    ['/debug js @odin-bot'],
    ['/debug rb @odin-bot'],
    [' /debug js @odin-bot'],
    [' /debug rb @odin-bot'],
  ])("'%s' triggers the callback", (string) => {
    expect(command.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['/debog'],
    ['debugs'],
    ['/dbug'],
    ['/de bug'],
    ['/debugging'],
    ['```function("/de bug", () => {}```'],
    ['/Debug'],
    [''],
    [' '],
    ['/ '],
    ['@odin-bot / debug'],
    ['/de&b^ug'],
    ['/d^bug'],
    ['@odin-bot/ debug'],
  ])("'%s' does not trigger the callback", (string) => {
    expect(command.regex.test(string)).toBeFalsy();
  });

  it.each([
    ['how about a flavour of /debug'],
    ['i think /debug is great'],
    ['hey @odin-bot, teach me to /debug'],
    ['/@odin-bot ^ /me /debug /test$*'],
  ])("'%s' - the command can be anywhere in the string", (string) => {
    expect(command.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['@user/debug'],
    ['watchaknowabout/debug'],
    ['/debugsgrealkdmsfalnd'],
    ['/debug/'],
    ['/debug*'],
    ['/debug...'],
  ])(
    "'%s' - command should be its own word/group - no leading or trailing characters",
    (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    },
  );
});

describe('/debug', () => {
  it('should return the correct output', () => {
    expect(command.cb({ content: '/debug' })).toMatchSnapshot();
  });
});

describe('/debug js', () => {
  it('should return the correct output', () => {
    expect(command.cb({ content: '/debug js' })).toMatchSnapshot();
  });
});

describe('/debug rb', () => {
  it('should return the correct output', () => {
    expect(command.cb({ content: '/debug rb' })).toMatchSnapshot();
  });
});
