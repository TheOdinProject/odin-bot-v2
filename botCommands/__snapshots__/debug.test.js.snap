// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!debug callback should return the correct output 1`] = `
Object {
  "embeds": Array [
    Object {
      "color": 13407555,
      "description": "Based on the description of your problem, you can get to the root of it using a debugger. Learning how to track down problems like this is an inevitable part of being a developer.",
      "fields": Array [
        Object {
          "name": "Javascript",
          "value": "https://developers.google.com/web/tools/chrome-devtools/javascript",
        },
        Object {
          "name": "Ruby",
          "value": "https://www.theodinproject.com/lessons/ruby-debugging#debugging-with-pry-byebug",
        },
      ],
      "title": "DEBUGGING",
      "url": "https://en.wikipedia.org/wiki/Debugging",
    },
  ],
}
`;
